pandas/core/frame.py

```diff
@@ -4867,10 +4867,34 @@ def _combine_const(self, other, func, errors='raise', try_cast=True):
         if lib.is_scalar(other) or np.ndim(other) == 0:
             return ops.dispatch_to_series(self, other, func)
 
-        new_data = self._data.eval(func=func, other=other,
-                                   errors=errors,
-                                   try_cast=try_cast)
-        return self._constructor(new_data)
+        elif (np.ndim(other) == 1 and isinstance(other, np.ndarray) and
+              len(other) == len(self.columns)):
+            right = np.broadcast_to(other, self.shape)
+            return ops.dispatch_to_series(self, right, func)
+
+        elif (np.ndim(other) == 1 and
+              isinstance(other, (tuple, np.ndarray)) and
```

Comment:
There is a test for that goes through this path with a tuple, but none with a list.  IIRC, no tests fail if list is included here.

Similarly, in the case above, only np.ndarray cases are tested.  Should tuple and/or list be allowed?